test/integration/instance.validations.test.ts

```diff
@@ -0,0 +1,552 @@
+      expect(err.errors).to.have.length(1);
+      if (getTestDialect() === 'ibmi') {
+        expect(err.errors[0].message).to.include('Duplicate key value specified');
+      } else {
+        expect(err.errors[0].path).to.include('uniqueName');
+        expect(err.errors[0].message).to.include('must be unique');
+      }
+    });
+
+    if (getTestDialect() !== 'ibmi') {
+      it('should allow a custom unique constraint error message', async () => {
+        class Book extends Model<InferAttributes<Book>, InferCreationAttributes<Book>> {
+          declare id: CreationOptional<number>;
+          declare uniqueName: string;
+        }
+
+        Book.init({
+          id: {
+            type: DataTypes.INTEGER,
+            primaryKey: true,
+            autoIncrement: true,
+          },
+          uniqueName: {
+            type: DataTypes.STRING,
+            unique: {
+              name: 'name',
+              msg: 'custom unique error message',
+            },
+          },
+        }, { sequelize });
+
+        const records = [
+          { uniqueName: 'unique name one' },
+          { uniqueName: 'unique name two' },
+        ];
+        await Book.sync({ force: true });
+        const instance0 = await Book.create(records[0]);
+        expect(instance0).to.be.ok;
+        const instance = await Book.create(records[1]);
+        expect(instance).to.be.ok;
+        const err = await expect(Book.update(records[0], { where: { id: instance.id } })).to.be.rejected;
+        expect(err instanceof ValidationError);
+        expect(err.errors).to.have.length(1);
+        expect(err.errors[0].path).to.include('uniqueName');
+        expect(err.errors[0].message).to.equal('custom unique error message');
+      });
+
+      it('should handle multiple unique messages correctly', async () => {
+        class Book extends Model<InferAttributes<Book>, InferCreationAttributes<Book>> {
+          declare uniqueName1: string;
+          declare uniqueName2: string;
+        }
+
+        Book.init({
+          uniqueName1: {
+            type: DataTypes.STRING,
+            unique: {
+              name: 'uniqueName1',
+              msg: 'custom unique error message 1',
+            },
+          },
+          uniqueName2: {
+            type: DataTypes.STRING,
+            unique: {
+              name: 'uniqueName2',
+              msg: 'custom unique error message 2',
+            },
+          },
+        }, { sequelize });
+
+        const records = [
+          { uniqueName1: 'unique name one', uniqueName2: 'unique name one' },
+          { uniqueName1: 'unique name one', uniqueName2: 'this is ok' },
+          { uniqueName1: 'this is ok', uniqueName2: 'unique name one' },
+        ];
+        await Book.sync({ force: true });
+        const instance = await Book.create(records[0]);
+        expect(instance).to.be.ok;
+        const err0 = await expect(Book.create(records[1])).to.be.rejected;
+        expect(err0 instanceof ValidationError);
```

Comment:
use `assert` when using instanceof like this, otherwise `err0`'s type will be `any`, not `ValidationError`

Same comment applies many times in this file